core/domain/change_domain.py

```diff
@@ -246,16 +248,16 @@ def validate_dict(
         if not valid_cmd_attribute_specs:
             raise utils.ValidationError('Command %s is not allowed' % cmd_name)
 
-        # Here we are deleting the 'name' key which cause MyPy to throw error,
-        # because MyPy does not allow key deletion from TypedDict. So to silent
-        # the error, we added an ignore here.
+        # Here we use MyPy ignore because we are deleting the 'name' key which
+        # cause MyPy to throw error, because MyPy does not allow key deletion
+        # from TypedDict. So to silent the error, we added an ignore here.
```

Comment:
> silent --> silence

 Done

> Actually I think this is a bit of a problem ...

`name` refers to cmd_name so I don't think we should remove this, also `name` is removed only in this place and here we are validating the dict.

https://github.com/oppia/oppia/blob/c2be7ec34a04f79f46304aae4109ba685da9d1b9/core/domain/exp_domain.py#L324


>  no one will know what the schema of the TypedDict is supposed to be when they look at it in the codebase

I think these are the only 2 places in the codebase where we are removing the name. And if we add proper comments then it will clear the things for someone who is looking into the codebase.